target/linux/ath79/dts/ar7242_meraki_mr12.dts

```diff
@@ -0,0 +1,180 @@
+#include "ar7242.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "meraki,mr12", "qca,ar7242";
+	model = "Meraki MR12";
+
+	aliases {
+		led-boot = &led_power_orange;
+		led-failsafe = &led_power_orange;
+		led-running = &led_power_green;
+		led-upgrade = &led_power_orange;
+	};
+
+	extosc: ref {
+		compatible = "fixed-clock";
+		#clock-cells = <0>;
+		clock-output-names = "ref";
+		clock-frequency = <40000000>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		wifi1 {
+			label = "green:wifi1";
+			gpios = <&gpio 11 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi2 {
+			label = "green:wifi2";
+			gpios = <&gpio 12 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi3 {
+			label = "green:wifi3";
+			gpios = <&gpio 13 GPIO_ACTIVE_LOW>;
+		};
+
+		wifi4 {
+			label = "green:wifi4";
+			gpios = <&gpio 14 GPIO_ACTIVE_LOW>;
+		};
+
+		wan {
+			label = "green:wan";
+			gpios = <&gpio 15 GPIO_ACTIVE_LOW>;
+		};
+
+		led_power_orange: power_orange {
+			label = "orange:power";
+			gpios = <&gpio 16 GPIO_ACTIVE_LOW>;
+			panic-indicator;
+		};
+
+		led_power_green: power_green {
+			label = "green:power";
+			gpios = <&gpio 17 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys";
+
+		reset {
+			label = "reset";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 8 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&pcie {
+	status = "okay";
+
+	ath9k: wifi@0,0 {
```

Comment:
the "ath9k" label is not being used

Reply:
Fixed.

Reply:
oops I forgot, also get rid of `gpio-cells` and `gpio-controller` because you're not using it, all LEDs are using the `&gpio` node

Reply:
oops I also forgot...

first item in `wifi@0,0` is supposed to be a `compatible` value

see your kernel log for the PCI device ID, here is the docs

https://www.kernel.org/doc/Documentation/devicetree/bindings/net/wireless/qca,ath9k.txt

(btw post a gist of your boot log at some point)

Reply:
https://gist.github.com/Hurricos/d0f061ca311498552bd34343b0348078

Reply:
so the line from boot log tells you the device id

`[    2.325787] pci 0000:00:00.0: [168c:002a] type 00 class 0x028000`

so compatible for `wifi@0,0,0` is like this

`		compatible = "pci168c,002a";`

Reply:
Gotcha, added:
```diff
@@ -75,7 +75,8 @@
 &pcie {
        status = "okay";
 
-       wifi@0,0 {
+       wifi@0,0,0 {
+               compatible = "pci168c,002a";
                reg = <0x0000 0 0 0 0>;
                qca,no-eeprom;
                mtd-mac-address = <&config 0x66>;
```

Compatibles what, help confirm the device is what it's supposed to be?

Reply:
tells the ath9k driver what chip it is, as there might be hardware differences

other than that its a formal thing

Reply:
[Added wifi@0,0,0](https://github.com/openwrt/openwrt/pull/3634/commits/e783ef483365f08712c29411b2a73891e5cc2ae8#diff-ebe7e10623c0c66c5d4660984125f80f6ddb6439e5b52f53dc35dd70f48c14ceR80) as per [@mpratt14's comment](https://github.com/openwrt/openwrt/pull/3634#discussion_r532116771). Marking as resolved.